globals.gd

```diff
@@ -767,23 +758,13 @@ class progress:
 		globals.state.allnpcs.append(person)
 	
 	func findnpc(newid):
-		var rval
-		var foundslave = false
-		if !allnpcs.empty():
```

Comment:
Is a null check not needed? AllNPC doesn't populate with values automatically

Reply:
The line "if !allnpcs.empty():" was not a null check, it was an empty check. If it was null then ".empty()" would produce errors, but allnpc is always an array. Using a for loop on an empty container is safe as it iterates zero times thus never executes the inner code. This can be easily tested by writing a demo case in globals._init().
The line `if npc.id != null:` is a null check, but it is a meaningless check as we should never been looking for a null id or "Null" id as the next line does a string comparison upon the same value.